docs/sources/flow/reference/components/module.string.md

```diff
@@ -0,0 +1,136 @@
+---
+title: module.string
+---
+
+# module.string
+
+`module.string` is a *Module loader* component. A *Module loader* is a Grafana Agent Flow 
+component which retreives a module and runs the components defined inside of it.
+
+*TODO: Add link to module concept page above once merged*
+
+## Usage
+
+```river
+module.string "LABEL" {
+	content   = CONTENT
+	arguments = {
+		argument1 = ARGUMENT1,
+		argument2 = ARGUMENT2,
+		...
+	}
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`content`   | `secret`    | The contents of the module to load as a secret string. | | yes
+`arguments` | `map(any)`  | The values for the supported arguments in the module contents. | | no
+
+`content` is a string that contains all the arguments, exports and components for the module. 
+`content` is typically loaded via the exports of another component. For example,
+
+- local.file.[label].content
+- remote.http.[label].content
+- remote.s3.[label].content
+
+`arguments` can contain, but are not limited to strings, components and component exports.
```

Comment:
I think this needs a few more pieces of detail, primarily: 

* Any argument marked non-optional in the module being loaded must be present in the `arguments` object 
* It is invalid to provide an argument that is not defined as an `argument` block in the loaded module. 

As it is now, it's not clear how you would know what to put in `arugments`. 

Also: I'm not sure we have to say what type of values are supported, we can just say it can be set to any River expression.